### PR TITLE
Adding inline adding of title under experiment.

### PR DESCRIPTION
**Pull request by liechtym**
### Changes

- Adding CSS for inline adding errors
- Adding AJAX capability for saving title.
- Add `inlineAddingEnabled` property to allow turning inline adding on and off (for experiment)
- Adding event listeners and handlers for tapping on "add title", inputting text, and handling blur, pressing enter, and pressing exit.
- Tweaking other elements to account for the `_titleInputActive` state.
- Add `inline-adding-enabled` property to one example element.

### CODE REVIEW

File: birch-artifact-grid-item.html [hunk 1]
```diff
@@ -14,6 +16,13 @@
   <link rel="import" type="css" href="birch-artifact-grid-item.css">
   <template>
     <wc-i18n-src locale='[[lang]]' locale-dir='[[localeDir()]]' domain='birch-artifact-grid-item'></wc-i18n-src>
+    <iron-ajax id="postTitle"
+      url="[[ajaxBase]]/artifactmanager/artifacts/[[data.id]]"
+      headers='[[authHeaders]]'
+      method="POST"
+      content-type="application/json"
```
Comment (jpodwys) [inline birch-artifact-grid-item.html]: `oak-ajax-behavior` already does this for you.

File: birch-artifact-grid-item.html [hunk 2]
```diff
@@ -204,14 +221,34 @@
...
     ],
     listeners: {
       'artifact.tap': '_goToItem'
     },
+    _titleLinkOverride: function(e) {
+      if (this._titleError) {
+        e.preventDefault();
+        this._titleError = false;
+        // The title class variable helps determine whether the contributor or normal title should be used
+        var titleClass = (this.view == 'my-favorites' || Number(this.view)) ? '.title-2' : '.title';
```
Comment (chadeddington) [inline birch-artifact-grid-item.html]: Don't need anymore

File: birch-artifact-grid-item.html [hunk 3]
```diff
@@ -260,16 +297,33 @@
...
-    _hideAddTitle: function(readOnly, title, dataOwner, inputtedOwner) {
-      if (readOnly || title) return true;
+    _handleEnterAndBlur: function(e) {
+      if ((e.keyCode == 13 || e.key == "Enter" || e.type == "blur") && e.currentTarget.value.length > 0) {
+        this._saveTitle(e.currentTarget.value);
+      } else if ((e.keyCode == 13 || e.key == "Enter" || e.type == "blur") && e.currentTarget.value.length == 0) { 
+        this._titleInputActive = false;
+      }
+    },
+    _handlePost_TitleError:function(e) {
```
Comment (chadeddington) [inline birch-artifact-grid-item.html]: UNDERSCORE??

File: birch-artifact-grid-item.html [hunk 4]
```diff
@@ -14,6 +16,12 @@
   <link rel="import" type="css" href="birch-artifact-grid-item.css">
   <template>
     <wc-i18n-src locale='[[lang]]' locale-dir='[[localeDir()]]' domain='birch-artifact-grid-item'></wc-i18n-src>
+    <iron-ajax id="postTitle"
+      url="[[ajaxBase]]/artifactmanager/artifacts/[[data.id]]"
+      headers='[[authHeaders]]'
+      method="POST"
+      on-response="_handlePostTitleResponse"
+      on-error="_handlePost_TitleError"></iron-ajax>
```
Comment (chadeddington) [inline birch-artifact-grid-item.html]: UNDERSCORE??

File: birch-artifact-grid-item.html [hunk 5]
```diff
@@ -311,9 +365,38 @@
...
+      }
+    },
     _stopProp: function(e) {
       if (!this.selectMode) e.stopPropagation();
     },
+    _toggleAddTitleInput: function(e) {
+      if (this.inlineAddingEnabled) {
+        if (e) e.preventDefault();
+        this._titleInputActive = true;
+        Polymer.dom.flush();
```
Comment (chadeddington) [inline birch-artifact-grid-item.html]: ?

**Pull request merged**